ShowHiddenChannels: better ui, alternative display mode

Nuckyz opened the pull request.

Nuckyz commented:
Found an issue that I will fix later

Nuckyz commented:
Ready

Vendicated commented on src/plugins/showHiddenChannels.tsx:
```diff
@@ -23,125 +23,177 @@ import { Flex } from "@components/Flex";
     options: {
         hideUnreads: {
-            description: "Hide unreads",
+            description: "Hide Unreads",
```
whats the point of this setting btw, why would you ever want to see unreads in a hidden channel

Nuckyz replied:
dunno, I didn't implement it

Vendicated reviewed (approved):
have you tested that Message Logger still works?

Nuckyz commented:
> have you tested that Message Logger still works?

not exactly since they were fairly simple changes

Delphox commented:
Hey something on this PR broke the Hide Muted Channels hiding muted hidden channels.

Just tested on Vencord 8a43e9b and it still works:

https://user-images.githubusercontent.com/5883538/214331072-4a0ec48e-8d58-4dc5-84cf-154cb1129cb4.mp4

However on Vencord 75050e7 and above it doesn't:

https://user-images.githubusercontent.com/5883538/214331439-d81a1a7b-e143-407a-899a-46f401b8c488.mp4

Nuckyz commented:
Fixed in

> Hey something on this PR broke the Hide Muted Channels hiding muted hidden channels.
> 
> Just tested on Vencord [8a43e9b](https://github.com/Vendicated/Vencord/commit/8a43e9b25f97cda19a5dc85020152020873c6a8e) and it still works:
> 
>  thorium_uJTBWscy6K.mp4 
> However on Vencord [75050e7](https://github.com/Vendicated/Vencord/commit/75050e74ca01856f9b48867d7945e2b7263b42ae) and above it doesn't:
> 
>  thorium_M64HvKHB59.mp4

Fixed in #454
